### PR TITLE
[Github Actions] Trigger Github Actions tests (`run-tests.yml` workflow) from a comment in a PR

### PR DESCRIPTION
**Pull request by valeriupredoi**
## Description

This PR enables the user to turn on running Github Actions tests (the full tests, from `run-tests.yml` workflow) in a Pull Request, just by commenting a string - tagging the bot @runGAtests - before merging; this is very useful since we don't want to add branches to the GA workflow, then remove, forget about removing them etc.

This is a PITA to test since GH wants these workflows to first be on the default branch (`main` in our case) so I emulated the behavior on my fork:

- I have the relevant workflow script in my `main` https://github.com/valeriupredoi/ESMValCore/blob/main/.github/workflows/run-tests-comment.yml
- then I open a PR and comment "@runGAtests" https://github.com/valeriupredoi/ESMValCore/pull/2
- then these little critters go about running https://github.com/valeriupredoi/ESMValCore/actions

Doc: https://esmvaltool--1520.org.readthedocs.build/projects/ESMValCore/en/1520/contributing.html#automated-testing

***

## [Checklist](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#checklist-for-pull-requests)

It is the responsibility of the author to make sure the pull request is ready to review. The icons indicate whether the item will be subject to the [🛠 Technical][1] or [🧪 Scientific][2] review.

[1]: https://docs.esmvaltool.org/en/latest/community/review.html#technical-review
[2]: https://docs.esmvaltool.org/en/latest/community/review.html#scientific-review

- [x] [🧪][2] The new functionality is [relevant and scientifically sound](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#scientific-relevance)
- [x] [🛠][1] This pull request has a [descriptive title and labels](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-title-and-label)
- [x] [🛠][1] All [checks below this pull request](https://docs.esmvaltool.org/projects/ESMValCore/en/latest/contributing.html#pull-request-checks) were successful

***

To help with the number pull requests:

- 🙏 We kindly ask you to [review](https://docs.esmvaltool.org/en/latest/community/review.html#review-of-pull-requests) two other [open pull requests](https://github.com/ESMValGroup/ESMValCore/pulls) in this repository
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -

### CODE REVIEW

**Comment by zklaus** [comment]
Really cool concept! :beers: 
I wonder if we should invite everyone to use it just yet. This has the potential to create a lot of load on the GA nodes with often limited utility.

When it is needed, it's a great tool that will simplify things a lot, but that is mainly for really advanced debugging situations or changes to the CI or similar, no? So my feeling is we should absolutely have it, but not tell people so casually to switch it on "just in case". Thoughts, @ESMValGroup/esmvaltool-coreteam?

**Comment by valeriupredoi** [comment]
thamks, Klaus! Yeah I was a bit on the fence myself about the widespread of its use that's why I added it only in the PR template - I can remove it from there too if peeps think this is to be kept under the wraps :+1:

**Comment by bouweandela** [comment]
Maybe not in the pull request checklist, it is long enough as it is already. This is definitely a function that the @ESMValGroup/tech-reviewers should be aware of though. Could you mention it in the documentation, e.g. in the section on [automated testing](https://docs.esmvaltool.org/projects/esmvalcore/en/latest/contributing.html#automated-testing)?

**Comment by valeriupredoi** [comment]
good idea @bouweandela - just did that, cheers :beer:

**Comment by valeriupredoi** [comment]
I added Milestone 2.5.0 tentatively, but am not in a rush :+1:

**Comment by schlunma** [comment]
If we can merge this before (or right after) https://github.com/ESMValGroup/ESMValCore/pull/1448 we can include it, but otherwise I would postpone it to v2.6. I would like to publish rc5 as fast as possible :+1:

**Comment by valeriupredoi** [comment]
okidoki, no rush from me, it doesn't depend on anything - rc5 time now :beer:

**Comment by schlunma** [comment]
@zklaus Do you have a minute now to briefly look over this? I think we can merge https://github.com/ESMValGroup/ESMValCore/pull/1448 very soon (just running some tests now), and afterwards I want to initialize rc5. If not, no problem, then we move this to v2.6. Cheers!

File: .github/pull_request_template.md
Comment (zklaus) [reply .github/pull_request_template.md:48]: Trailing space...

**Review by zklaus — APPROVED** [review]

**Comment by schlunma** [comment]
Thanks Klaus! :tada:

**Review by schlunma — APPROVED** [review]

**Comment by schlunma** [comment]
@runGAtests

**Comment by schlunma** [comment]
Is this features supposed to work in this PR? Or can it only be used afterwards?

**Comment by valeriupredoi** [comment]
only after merge, when the workflow resides on the default (main) branch

**Comment by valeriupredoi** [comment]
thanks Manu!

**Pull request merged**
